drivers/sensor/hx711/hx711.c

```diff
@@ -0,0 +1,562 @@
+ * @param dev Pointer to the hx711 device structure
+ * @param val sensor_value struct. Only val1 is used.
+ *            valid values are :
+ *               HX711_RATE_10HZ (default),
+ *               HX711_RATE_80HZ
+ *
+ * @retval 0 on success,
+ * @retval -ENOTSUP error if an invalid rate value is passed.
+ *
+ */
+static int hx711_attr_set_rate(struct hx711_data *data,
+			       const struct sensor_value *val)
+{
+	int ret;
+
+	switch (val->val1) {
+	case HX711_RATE_10HZ:
+	case HX711_RATE_80HZ:
+		data->rate = val->val1;
+		ret = gpio_pin_set(data->rate_gpio,
+				   hx711_config.rate_pin,
+				   data->rate);
+		return ret;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+/**
+ * @brief Set HX711 offset.
+ *
+ * @param dev Pointer to the hx711 device structure
+ * @param offset sensor_value struct. Offset used to calculate the real weight.
+ *        Formula is :
+ *          weight = slope * (reading - offset)
+ *          Only val1 is used.
+ *
+ * @return void
+ *
+ */
+static void hx711_attr_set_offset(struct hx711_data *data,
+				  const struct sensor_value *offset)
+{
+	data->offset = offset->val1;
+}
+
+/**
+ * @brief Set HX711 offset.
+ *
+ * @param dev Pointer to the hx711 device structure
+ * @param slope sensor_value struct. Slope used to calculate the real weight.
+ *        Formula is :
+ *          weight = slope * (reading - offset)
+ *
+ * @return void
+ *
+ */
+static void hx711_attr_set_slope(struct hx711_data *data,
+				 const struct sensor_value *slope)
+{
+	data->slope.val1 = slope->val1;
+	data->slope.val2 = slope->val2;
+}
+
+/**
+ * @brief Set HX711 attributes.
+ *
+ * @param dev Pointer to the hx711 device structure
+ * @param chan Channel to set.
+ *             Supported channels :
+ *               SENSOR_CHAN_WEIGHT
+ *               SENSOR_CHAN_ALL
+ * @param attr Attribute to change.
+ *             Supported attributes :
+ *               SENSOR_ATTR_SAMPLING_FREQUENCY
+ *               SENSOR_ATTR_OFFSET
+ *               SENSOR_ATTR_SLOPE
+ *               SENSOR_ATTR_GAIN
+ * @param val   Value to set.
+ * @retval  -ENOTSUP if an invalid attribute is given
+ *
+ */
+static int hx711_attr_set(const struct device *dev,
+			  enum sensor_channel chan,
+			  enum sensor_attribute attr,
+			  const struct sensor_value *val)
+{
+	int ret = 0;
+	struct hx711_data *data = dev->data;
+
+	switch (attr) {
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		ret = hx711_attr_set_rate(data, val);
+		if (ret == 0) {
+			LOG_DBG("Attribute RATE set to %d\n", data->rate);
+		}
+		return ret;
+	case SENSOR_ATTR_OFFSET:
+		hx711_attr_set_offset(data, val);
+		LOG_DBG("Attribute OFFSET set to %d\n", data->offset);
+		return ret;
+	case SENSOR_ATTR_SLOPE:
+		hx711_attr_set_slope(data, val);
+		LOG_DBG("Attribute SLOPE set to %d.%d\n",
+			data->slope.val1,
+			data->slope.val2);
+		return ret;
+	case SENSOR_ATTR_GAIN:
+		if (hx711_attr_set_gain(dev, val) == 0) {
+			LOG_DBG("Attribute GAIN set to %d\n", data->gain);
+		}
+		return ret;
```

Comment:
Please update **ret** after hx711_attr_set_gain(). It invokes hx711_sample_fetch() and it's possible to change it in case of failure.

Reply:
Right.

Reply:
done.